source/blog/_posts/2016-08-26-css-coding-guidelines.md

```diff
@@ -0,0 +1,340 @@
+You can learn more on how to use the helpers in this [comprehensive guide](https://developers.shopware.com/designers-guide/less/).
+
+## Think in objects
+
+To write modular CSS you have to push yourself to think in objects. What are objects in this case? Objects are small
+independent parts of functionality. UI elements like headers, footers, buttons or form elements are objects.
+ 
+Object are **always** declared using a class selector. ID & tag based selectors not that easy to customize. For example
+to override an ID based selector you have to declare a selector chain with 255 classes to override one ID.
+
+```
+.button {
+    background: linear-gradient(#eee, #ccc);
+    border: 1px solid #999;
+    color: #333;
+    cursor: pointer;
+    padding: 1em 1.5em;
+     
+    &:hover {
+        background: linear-gradient(#fff, #ddd);
+        color: #111;
+    }
+}
+```
+
+The `&` selector in LESS makes it easy to define different "states" of object. In the above example we define the
+"hover" state for the button.
+
+## Parent-child relationshop
+Using a consistent naming scheme to define the parent-child relationship between objects helps to write much cleaner 
+CSS code. Here's an example:
+
+```
+/** LESS code */
+.post {
+    margin: 2em;
+     
+    .title {
+        font-size: 2em;
+        font-weight: normal;
+    }
+}
+  
+/** Processed result */
+.post {
+    margin: 2em
+}
+.post .title {
+    font-size: 2em;
+    font-weight: normal;
+}
+```
+
+As you can see in the above example we came up a hard relationship parent element `.post` & the child element `.title`.
+The styling of the title will only be applyed when the parent element has the class `.post`. That means we don't have the
+chance to use the post title in a modified version for another object. So let's try to get rid of the hard relationship
+between these two selector:
+
+```
+/** LESS code and processed result */
+.post {
+    margin: 2em;
+}
+.post--title {
+    font-size: 2em;
+    font-weight: normal;
+}
+```
+
+The main benefit of this approach is that the defined styles can be used throughout the application & we don't have
+to worry about conflicts with similar named objects.
+
+## Subclassing objects
+The most object oriented concept are using the concept of extending classes from other objects. This technique is called
+*"subclassing"*. THis is useful when you want to extend the properties of the object & define additional properties.
+
+Let's go back to the button example & extend it to a drop down button:
+
+```
+.button {
+    background: linear-gradient(#eee, #ccc);
+    border: 1px solid #999;
+    color: #333;
+    cursor: pointer;
+    padding: 1em 1.5em;
+     
+    &:hover {
+        background: linear-gradient(#fff, #ddd);
+        color: #111;
+    }
+}
+  
+.dropdown--button {
+    &::after {
+        content: '&#9662;'
+    }
+}
+```
+
+In this example we extended the button & transformed it to a drop down button. To use this button styles we have to
```

Comment:
Bin mir nicht ganz sicher, aber hier ggf. "To use **those** button styles [...]"